### PR TITLE
Lambda: create list_aliases functionality

### PR DESCRIPTION
**Pull request by freshollie**
- Creates the [list_aliases](https://boto3.amazonaws.com/v1/documentation/api/1.26.85/reference/services/lambda/client/list_aliases.html) functionality
- Redesign internal storage of aliases to correct mistaken implementation (Duplicate alias names could be created under the same lambda pointing to different versions)
- Adds tests for new functionality
- Worth noting I've not included functionality to filter by "version", can be implemented if needed

### CODE REVIEW

**Comment by freshollie** [comment]
Let me know if you want docs updated as part of this PR

File: moto/awslambda/models.py [hunk 1]
```diff
@@ -954,43 +953,6 @@ def _create_zipfile_from_plaintext_code(code: str) -> bytes:
     def delete(self, account_id: str, region: str) -> None:
         lambda_backends[account_id][region].delete_function(self.function_name)
 
-    def delete_alias(self, name: str) -> None:
```
Comment (freshollie) [inline moto/awslambda/models.py:957]: Moved the logic to lambda storage level

File: moto/awslambda/models.py [hunk 2]
```diff
@@ -1205,22 +1167,34 @@ def __init__(self, region_name: str, account_id: str):
         self.region_name = region_name
         self.account_id = account_id
 
+        # function-arn -> alias -> LambdaAlias
+        self._aliases: Dict[str, Dict[str, LambdaAlias]] = defaultdict(lambda: {})
```
Comment (freshollie) [inline moto/awslambda/models.py:1171]: Aliases are associated with the "lambda" not the "lambda version". We can't store these in "lambda version" as it makes it very complex to check if the alias already exists before writing. It also simplifies fetching a list of aliases as we don't need to iterate over each "version" to find it's alias

**Comment by freshollie** [comment]
Will look into the failing test

**Review by bblommers — CHANGES_REQUESTED** [review]
This looks really good. Thank you for this PR @freshollie!

To show our thanks, we'd like to share some of the donations that we've received with you. PR's like this are the big reason that Moto is as successful as it is - so it's only fair that you, as a contributor, get to share the spoils.

We've created a companion website with more information:
https://payments.getmoto.org/

This website is still in beta, but everything should work. Feel free to open a bug or discussion if you run into any problems:
https://github.com/getmoto/payments

**Pull request merged**